### ISSUE TITLE
Customized POS Module has a Customer/Tax Error

### ISSUE THREAD

**Issue by bkm888** [mntechnique/jmi_erpnext#56]
An error in "when to apply" tax occurs in our Customized POS module when a default customer is used.

Currently all tax rules refer to the Billing County in the customer address to determine which tax to apply. In the POS Profiles of the Mobile Sales users the "Taxes and Charges" field is left blank so that the tax rules that read the Billing County operate correctly. For our Mobile Sales users this is working just fine.

For the fixed store locations, the Default Customer was created to handle all of the anonymous walk-in customer traffic. The Default Customer was then given an address equal to the store location so the tax rule would be correct. In the POS profile for the fixed stores the Default Customer for each store is named in the "Customer" field of the profile. 

When the Default Customer is named in the POS profile, the POS module is not checking the customer billing address in order to determine the tax rule to apply. In this case NO TAX is being applied. If we remove the Default Customer from the "Customer Field" in the POS profile and manually select the default customer in the POS module at the start of each sale, the correct tax rules are used and the taxes are applied.

As a temporary fix we have FORCED the tax rule for the store County location to be used by adding the County Name into the "Taxes and Charges" field in the POS profile for the store.

This temporary fix has generated a additional problems. When one of our commercial accounts comes into the store to get quick supplies, they are charged the local store tax even if they are from a different county or tax exempt. So when we delete the Default Customer from the POS screen at the time of sale and add the commercial customer, the POS Profile for the store is overriding the customers county tax rule with it's own local tax taken from the "Taxes and Charges" field of the profile.

In short, we need for the Default Customer to work properly when using the POS module. It must look at the customer billing county just like all of the others.

This does not happen when using the Sales Invoice, it is only in the Customized POS module.

Please investigate and let me know what your ideas are to resolve.

**Comment by unmesh87** [comment]
@bkm888 

We tried replicating the issue on our GCP staging server by changing customer through Point of sale setup and seems to be working fine.

Observation- 

- Tax rule for some county might be missing. Need to identify such missing data and apply correct tax rule.

- Or county on customer address is not mentioned correctly. 

Reference-

- On GCP test server, check Sales invoice: RDU-103227

Please run the test again and update if better fix is expected.

**Comment by bkm888** [comment]
Unmesh,
 
This issue (Github #56) has to do with the Default Customer defined in the POS Profile when it is used in POS.
 
The example invoice you have chosen for your example is incorrect. You chose to use a regular commercial customer instead of the default customer (RDU Walkin).
 
I have changed the password to the user rdustore@majesticsolutions.com to "password" to make it easier for us to test this.
 
Please have a look again. When you use POS and the default customer is applied from the POS Profile, the taxes are not working properly.
 
If you then go to the POS Profile and delete the default customer and save the profile. Next go back to the POS and this time select the "RDU Walkin" customer manually, the transaction taxes are working correctly.
 
This issue is specific to whatever customer is being defined as the default customer in the POS Profile. If you look into the "RDU Walkin" customer, you will find everything is defined correctly. If it were NOT then the POS tansaction where you enter that name manually would also be broken, but it is not. It only breaks if the customer is defined as the default customer in the POS profile.
 
Please see example Sales Invoice RDU-103228 on the GCP test server. Then try making some of your own, but limit the customer to the RDU Walkin customer.
 
It should not act differently if it is manually entered or if it is the derfault enterd by the POS profile, yet that is exactly the problem.
 
Bryan Miles
JMI Barcodes
bkm@jmibarcodes.com
 
 
-----Original Message-----

**Comment by bkm888** [comment]
Maybe a better explanation of this is due.

I may have stated the original problem incorrectly, or less clear.

If the default customer (RDU Walkin) is defined in the POS profile, the taxes triggered by RDU Walkin's Billing County do not appear to be working on the POS screen. 

If the POS profile has the default customer removed so the field is empty, and a POS transaction is made using the RDU Walkin customer, the Billing County of RDU Walkin properly triggers the taxes and they appear on the POS screen total.

Hope that simplifies it.

BKM

**Comment by unmesh87** [comment]
@bkm888 

Thanks for simplifying the issue. We had different cases run through and found if; default customer on POS profile is populated, sales template is applied along with tax rule then taxes are correctly picked up on POS screen. When tax rule alone for default customer is set, the **Grand Total** on POS screen do not pick up taxes but are applied correctly in Sales Invoice.

It seems a pull request oriented development work and should go in the core system for community to benefit at large and will be a separate project altogether. 

Would be helpful if we can move issue to [ERPNext github repo](https://github.com/frappe/erpnext/issues).

**Comment by bkm888** [comment]
Unmesh,
 
Thank you for evaluating this for me. If you want to move it to the ERPNext github account, then go ahead. However, I still need this fixed. PLease work out a path to fixing this for me and send me your estimate.
 
Thanks,
 
Bryan Miles
JMI Barcodes
bkm@jmibarcodes.com
 
-----Original Message-----

**Comment by unmesh87** [comment]
@bkm888 

Will do that, for now closing this issue.
